src/components/figures/chart/chartJs/chart_js_extension.ts

```diff
@@ -1,12 +1,24 @@
-import { Plugin as ChartJSPlugin } from "chart.js";
 import { Registry } from "../../../../registries/registry";
 
-export const chartJsExtensionRegistry = new Registry<ChartJSPlugin>();
+export const chartJsExtensionRegistry = new Registry<{
+  register: (chart: typeof window.Chart) => void;
+  unregister: (chart: typeof window.Chart) => void;
+}>();
 
-/** Return window.Chart, making sure all our extensions are loaded in ChartJS */
-export function getChartJSConstructor() {
-  if (window.Chart && !window.Chart?.registry.plugins.get("chartShowValuesPlugin")) {
-    window.Chart.register(...chartJsExtensionRegistry.getAll());
+export function registerChartJSExtensions() {
+  if (!window.Chart || window.Chart.registry.plugins.get("chartShowValuesPlugin")) {
+    return;
+  }
+  for (const registryItem of chartJsExtensionRegistry.getAll()) {
+    registryItem.register(window.Chart);
```

Comment:
```suggestion
    window.Chart.register(registryItem);
```
maybe?
and the registry would only be a list of plugins

Reply:
In 18.2 (or 18.3?) we need to register things that aren't plugins 🙃